pkg/pkg.bzl

```diff
@@ -187,25 +186,23 @@ def _pkg_tar_impl(ctx):
         )
 
     if ctx.attr.modes:
-        args += [
-            "--modes=%s=%s" % (_quote(key), ctx.attr.modes[key])
-            for key in ctx.attr.modes
-        ]
+        for key in ctx.attr.modes:
+            args.add("--modes", "%s=%s" % (_quote(key), ctx.attr.modes[key]))
```

Comment:
Could this loop and the ones below be replaced with `args.add_all`?  Might be something we could nicely do with a lambda in Bazel 5, if/when we want to break compatibility.

Reply:
`add_all` does not work for this case. It does something like 
`--modes a=b,c=d,....` but we need '--modes=a=b --modes=c=d ...`
The same for all the others.